benchmarks/helics/BenchmarkFederate.hpp

```diff
@@ -0,0 +1,320 @@
+
+    std::unique_ptr<helics::CombinationFederate>
+        fed; //<! the federate object to use in derived classes
+
+    // Command line options
+    std::unique_ptr<helics::helicsCLI11App> app; //<! the CLI11 app object to use in derived classes
+
+    // variables to track current state, mainly for gbenchmark piecewise setup
+    bool initialized{false};
+    bool readyToRun{false};
+
+    // functions to be overriden by derived benchmark classes
+    /** set/override default base parameter values before arguments are parsed, and modify CLI11 options for default arguments*/
+    virtual void setupArgumentParsing() {}
+    /** initialization steps after command line options have been parsed, but before the federate object is created
+     * @param fi a reference to a helics::FederateInfo object that can be used to change how the federate object is made
+     */
+    virtual void doParamInit(helics::FederateInfo& fi) { (void)fi; }
+    /** initialization that needs to happen after the federate object is created, such as creating endpoints and inputs*/
+    virtual void doFedInit() {}
+    /** initialization that requires the federation to be set up, but before timing starts, like creating initial events*/
+    virtual void doMakeReady() {}
+    /** the main loop for the benchmark*/
+    virtual void doMainLoop() {}
+    /** returns the federate name (should be unique within the federation*/
+    virtual std::string getName() { return ""; }
+    /** allows a federate to add custom results to get printed when running in standalone mode*/
+    virtual void doAddBenchmarkResults() {}
+
+  public:
+    BenchmarkFederate(): BenchmarkFederate("") {}
+
+    /** constructor taking a name for the benchmark app
+     * @param name the name of the benchmark federate, shown by CLI11 --help option
+     */
+    explicit BenchmarkFederate(const std::string& name):
+        benchmarkName(name),
+        app(std::make_unique<helics::helicsCLI11App>(name + " Benchmark Federate"))
+    {
+        addResult("BENCHMARK FEDERATE TYPE", "benchmark_federate_type", name);
+        app->allow_extras();
+
+        // add common time options (optional)
+        opt_delta_time =
+            app->add_option("--delta_time", deltaTime, "the sampling rate/timestep size");
+        opt_final_time =
+            app->add_option("--final_time", finalTime, "final/stop time for the benchmark");
+        opt_delta_time->ignore_underscore();
+        opt_final_time->ignore_underscore();
+
+        // add common index options (optional)
+        opt_index = app->add_option("--index", index, "the index of this phold federate");
+        opt_max_index =
+            app->add_option("--max_index", maxIndex, "the maximum index given to a phold federate");
+        opt_max_index->ignore_underscore();
+
+        // add a flag for printing system info
+        app->add_flag_function(
+            "--print_systeminfo",
+            [](int count) {
```

Comment:
I think you could use `add_flag_callback` here since count isn't really doing much and --print_system_info=false is likely to still trigger it since that would result in a -1 in this case.